lib/playgroundbook_renderer/playgroundbook_renderer.rb

```diff
@@ -0,0 +1,73 @@
+require 'colored'
+require 'pathname'
+require 'yaml'
+require 'fileutils'
+require 'playgroundbook_renderer/contents_manifest_generator'
+require 'playgroundbook_renderer/chapter_collator'
+
+module Playgroundbook
+  ContentsDirName = 'Contents'
+  ChaptersDirName = 'Chapters'
+
+  # A renderer for playground books.
+  class Renderer < AbstractLinter
+    attr_accessor :yaml_file_name
+    attr_accessor :contents_manifest_generator
+    attr_accessor :chapter_collator
+    attr_accessor :ui
+
+    def initialize(yaml_file_name, 
+        contents_manifest_generator = ContentsManifestGenerator.new, 
```

Comment:
This is currently passing in no parameters even though one is required. We need to move that parameter to the `generate!` function instead of the initializer.